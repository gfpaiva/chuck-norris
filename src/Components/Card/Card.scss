@import '../../Scss/root';

.card {
  @include shadow;
  background-color: $neutral-color;
  border-radius: $border-radius;
  padding: 2rem;

  &--hover {
    @include hover-shadow;
  }
}
