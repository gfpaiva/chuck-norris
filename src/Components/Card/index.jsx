import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import './Card.scss';

export default function Card({
  children,
  className,
  hover,
  href,
}) {
  const classNames = `card${hover ? ' card--hover' : ''}${className ? ` ${className}` : ''}`;

  return href ? (
    <Link className={classNames} to={href}>
      {children}
    </Link>
  ) : (
    <div className={classNames}>
      {children}
    </div>
  );
}

Card.defaultProps = {
  className: '',
  hover: false,
  href: '',
};

Card.propTypes = {
  hover: PropTypes.bool,
  href: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};
