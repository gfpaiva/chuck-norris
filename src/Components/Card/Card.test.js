import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import Card from './index';

describe('<Card />', () => {
  it('should mount properly', async () => {

    const wrapper = mount(
      <Card>
          Test
      </Card>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should mount link properly', async () => {

    const wrapper = mount(
      <Router>
        <Card href="/">
            Test
        </Card>
      </Router>
    );

    expect(wrapper.find('Card')).toMatchSnapshot();
  });

  it('should mount class variations properly', async () => {

    const wrapper = mount(
      <Card hover className="test">
          Test
      </Card>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
