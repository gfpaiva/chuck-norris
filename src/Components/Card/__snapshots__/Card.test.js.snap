// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should mount class variations properly 1`] = `
<Card
  className="test"
  hover={true}
  href=""
>
  <div
    className="card card--hover test"
  >
    Test
  </div>
</Card>
`;

exports[`<Card /> should mount link properly 1`] = `
<Card
  className=""
  hover={false}
  href="/"
>
  <Link
    className="card"
    to="/"
  >
    <a
      className="card"
      href="/"
      onClick={[Function]}
    >
      Test
    </a>
  </Link>
</Card>
`;

exports[`<Card /> should mount properly 1`] = `
<Card
  className=""
  hover={false}
  href=""
>
  <div
    className="card"
  >
    Test
  </div>
</Card>
`;
